rules/static.go

```diff
@@ -595,11 +603,39 @@ func (r *DefaultRuleRenderer) filterOutputChain(ipVersion uint8) *Chain {
 	// If we reach here, the packet is not going to a workload so it must be going to a
 	// host endpoint. It also has no endpoint mark so it must be going from a process.
 
-	if ipVersion == 4 && r.IPIPEnabled {
+	if ipVersion == 4 {
+		r.addIPIPVXLANEgressAllowRules(&rules)
+	}
+
+	// Apply host endpoint policy.
+	rules = append(rules,
+		Rule{
+			Action: ClearMarkAction{Mark: r.allCalicoMarkBits()},
+		},
+		Rule{
+			Match:  Match().NotConntrackState("DNAT"),
+			Action: JumpAction{Target: ChainDispatchToHostEndpoint},
+		},
+		Rule{
+			Match:   Match().MarkSingleBitSet(r.IptablesMarkAccept),
+			Action:  r.filterAllowAction,
+			Comment: []string{"Host endpoint policy accepted packet."},
+		},
+	)
+
+	return &Chain{
+		Name:  ChainFilterOutput,
+		Rules: rules,
+	}
+}
+
+func (r *DefaultRuleRenderer) addIPIPVXLANEgressAllowRules(rules *[]Rule) {
+
+	if r.IPIPEnabled {
 		// When IPIP is enabled, auto-allow IPIP traffic to other Calico nodes.  Without this,
 		// it's too easy to make a host policy that blocks IPIP traffic, resulting in very confusing
 		// connectivity problems.
-		rules = append(rules,
+		*rules = append(*rules,
```

Comment:
I'd normally fashion a function like this after `append`:
```
appendIPIPVXLANEgressAllowRules(rules []Rule) []Rule
```
Guess this is fine too but I don't often see pointers to slices in Go.